### ISSUE TITLE
Allow Regular Expressions in Jest Runner

### ISSUE THREAD

**Issue by RCiesielczuk** [wix/Detox#1081]
#### Description

Jest allows you to [provide a regex pattern](https://jestjs.io/docs/en/cli#jest-regexfortestfiles) in order to run tests. Unfortunately I am unable to get the tests to run when the regex pattern is passed to Jest.

In the Jest documentation it's mentioned that:

>Depending on your terminal, you may need to quote this argument: jest "my.*(complex)?pattern"

The proposed solution wraps the file argument in quotation marks to in order to support regex patterns along side specifying a file path.

#### Steps to Reproduce

1. Attempt to pass in a REGEX pattern via the `-f` flag and noticed that the test fails to run.
   - `npm run e2e:ios -- -f e2e/(01.santity)*.*`
2. If we wrap the `testFolder` argument within the quotation marks, then the test will run with the provided regex.

The proposed solution is to edit `runJest()` and wrap the test folder in quotation marks
```js
function runJest() {
  ...
  const testFolderString = `"${testFolder}"`
  const command = `${binPath} ${testFolderString} ${configFile}${color} --maxWorkers=${program.workers} ${platformString} ${collectExtraArgs()}`;
  ...
}
```

#### Detox, Node, Device, Xcode and macOS Versions

* Detox: 9.0.8
* React Native: 0.56.0
* Node: v10.13.0
* Device: iPhone X (iOS 12.0)
* Xcode: 10.0 (10A255)
* macOS: 10.14.1 (18B75)

**Comment by RCiesielczuk** [comment]
Hey @DanielMSchmidt,

Thanks for your feedback - I've submitted a PR (#1083) 😄

**Comment by noomorph** [comment]
Fixed in #1083.
